analyzer/metadata_registry.go

```diff
@@ -0,0 +1,117 @@
+package analyzer
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/iancoleman/strcase"
+	registry "github.com/oasisprotocol/metadata-registry-tools"
+	"github.com/oasisprotocol/oasis-indexer/config"
+	"github.com/oasisprotocol/oasis-indexer/log"
+	"github.com/oasisprotocol/oasis-indexer/metrics"
+	"github.com/oasisprotocol/oasis-indexer/storage"
+)
+
+const MetadataRegistryAnalyzerName = "metadata_registry"
+
+type MetadataRegistryAnalyzer struct {
+	qf       QueryFactory
+	target   storage.TargetStorage
+	logger   *log.Logger
+	metrics  metrics.DatabaseMetrics
+	interval time.Duration
+}
+
+var _ Analyzer = (*MetadataRegistryAnalyzer)(nil)
+
+func (a *MetadataRegistryAnalyzer) Name() string {
+	return MetadataRegistryAnalyzerName
+}
+
+func NewMetadataRegistryAnalyzer(cfg *config.AnalyzerConfig, target storage.TargetStorage, logger *log.Logger) (*MetadataRegistryAnalyzer, error) {
+	// Parse config
+	var interval time.Duration
+	var err error
+	if cfg.Interval == "" {
+		interval = 1 * time.Minute // default interval
+	} else {
+		interval, err = time.ParseDuration(cfg.Interval)
+		if err != nil {
+			logger.Error("error parsing analysis interval",
+				"err", err.Error(),
+			)
+			return nil, err
+		}
+	}
+	if cfg.ChainID == "" {
+		return nil, fmt.Errorf("metadata_registry analyzer: `ChainID` must be specified in the config")
+	}
+
+	logger.Info("Starting metadata_registry analyzer")
+	return &MetadataRegistryAnalyzer{
+		interval: interval,
+		qf:       NewQueryFactory(strcase.ToSnake(cfg.ChainID), "" /*runtime*/),
```

Comment:
do our queries use the chain ID?

Reply:
hm they do

Reply:
I was surprised too :)

Reply:
Yeah, this was both for "future proofing" chain ID context separated tables as well trying to make the current scheme somewhat reusable.